src/app/home/home.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit } from "@angular/core";
+import { HttpClass } from "../http.service";
+
+import { CovidInfo, CountryInfo } from "../interface/interface";
+
+@Component({
+  selector: "app-home",
+  templateUrl: "./home.component.html",
+  styleUrls: ["./home.component.scss"],
+})
+export class HomeComponent implements OnInit {
+  isLoading: boolean = false;
+  data: CovidInfo;
+  chartData: CountryInfo[];
+  error = null;
+
+  constructor(private http: HttpClass) {}
+
+  ngOnInit(): void {
+    this.isLoading = true;
```

Comment:
Loading should be coupled with the async calls (fetchingData, e.g.), if possible.